testhelper/tests.go

```diff
@@ -268,6 +268,12 @@ func (options *TestOptions) RunTestUpgrade() (*terraform.PlanStruct, error) {
 				if result != nil && resultErr == nil {
 					logger.Log(options.Testing, "Parsing plan output to determine if any resources identified for destroy (PR branch)..")
 					options.checkConsistency(result)
+					// Adding optional upgrade support on PR Branch
+					if options.ApplyUpgradePR {
+						logger.Log(options.Testing, "Validating Optional upgrade on Current Branch (PR):", cur.Name())
+						_, resultErr = terraform.InitAndApplyE(options.Testing, options.TerraformOptions)
```

Comment:
@ocofaigh Are we expecting the "Apply" after upgrade to error? Or were you looking for a consistency check for the apply? Terratest only returns the PlanStruct (that we use for our "checkConsistency") when we run a plan, we would have to research if we can return that data on an apply if we need to check consistency (if there are things replaced or destroyed).

Reply:
@toddgiguere The issue that was reported was only seen on apply. So we need to support doing the actual apply (only if the consistency test for the upgrade passes)

Reply:
Ok then we should be good with this Apply call as it is, but @imprateeksh we only need to run this final apply if the option is set AND there are no failures up to that point. You can check that by using `options.Testing.Failed()`

Reply:
@toddgiguere I have made the change as you suggested. 
Can you please review it once?